sklearn/utils/tests/test_extmath.py

```diff
@@ -416,13 +416,94 @@ def test_incremental_variance_update_formulas():
     old_means = X1.mean(axis=0)
     old_variances = X1.var(axis=0)
     old_sample_count = X1.shape[0]
-    final_means, final_variances, final_count = _batch_mean_variance_update(
-        X2, old_means, old_variances, old_sample_count)
+    final_means, final_variances, final_count = \
+        _incremental_mean_and_var(X2, old_means, old_variances,
+                                  old_sample_count)
     assert_almost_equal(final_means, A.mean(axis=0), 6)
     assert_almost_equal(final_variances, A.var(axis=0), 6)
     assert_almost_equal(final_count, A.shape[0])
 
 
+def test_incremental_variance_numerical_stability():
+    # Test Youngs and Cramer incremental variance formulas.
+
+    def np_var(A):
+        return A.var(axis=0)
+
+    # Naive one pass variance computation - not numerically stable
+    # https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance
+    def one_pass_var(X):
+        n = X.shape[0]
+        exp_x2 = (X ** 2).sum(axis=0) / n
+        expx_2 = (X.sum(axis=0) / n) ** 2
+        return exp_x2 - expx_2
+
+    # Two-pass algorithm, stable.
+    # We use it as a benchmark. It is not an online algorithm
+    # https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Two-pass_algorithm
+    def two_pass_var(X):
+        n = X.shape[0]
+        mean = X.mean(axis=0)
+        Y = X.copy()
+        var = np.array([(Y[i, :]-mean) ** 2 for i in range(n)]).sum(axis=0)
```

Comment:
This is the same as `np.mean((X - mean)**2, axis=0)` right? Is there any reason why you are iterating across samples?